keps/sig-apps/3850-backoff-limits-per-index-for-indexed-jobs/README.md

```diff
@@ -0,0 +1,1435 @@
+      - Impact of its outage on the feature:
+      - Impact of its degraded performance or high-error rates on the feature:
+-->
+
+### Scalability
+
+<!--
+For alpha, this section is encouraged: reviewers should consider these questions
+and attempt to answer them.
+
+For beta, this section is required: reviewers must answer these questions.
+
+For GA, this section is required: approvers should be able to confirm the
+previous answers based on experience in the field.
+-->
+
+###### Will enabling / using this feature result in any new API calls?
+
+No.
+
+<!--
+Describe them, providing:
+  - API call type (e.g. PATCH pods)
+  - estimated throughput
+  - originating component(s) (e.g. Kubelet, Feature-X-controller)
+Focusing mostly on:
+  - components listing and/or watching resources they didn't before
+  - API calls that may be triggered by changes of some Kubernetes resources
+    (e.g. update of object X triggers new updates of object Y)
+  - periodic API calls to reconcile state (e.g. periodic fetching state,
+    heartbeats, leader election, etc.)
+-->
+
+###### Will enabling / using this feature result in introducing new API types?
+
+No.
+
+<!--
+Describe them, providing:
+  - API type
+  - Supported number of objects per cluster
+  - Supported number of objects per namespace (for namespace-scoped objects)
+-->
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+
+No.
+
+<!--
+Describe them, providing:
+  - Which API(s):
+  - Estimated increase:
+-->
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+  Yes, but only when the `.spec.backoffLimitPerIndex` field is set.
+
+  - API type(s): Job
+  - Estimated increase in size:
+    - New `.status.failedIndexes` field in Status and `.status.completedIndexes`
+pre-existing field are impacted. When the scalability limits are respected,
+then the maximal increase of the total size of both fields can be estimated
+as `190Ki` (see [The Job object too big](#the-job-object-too-big) for more details),
+    - New `.spec.backoffLimitPerIndex` field of `*int32` is 12 bytes.
+
+  - API type(s): Pod
+  - Estimated increase in size:
+    the new annotation `batch.kubernetes.io/job-index-failure-count` to keep the
+    current number of retries per index. Is around 50 bytes.
+
+
+<!--
+Describe them, providing:
+  - API type(s):
+  - Estimated increase in size: (e.g., new annotation of size 32B)
+  - Estimated amount of new objects: (e.g., new Object X for every existing Pod)
+-->
+
+###### Will enabling / using this feature result in increasing time taken by any operations covered by existing SLIs/SLOs?
+
+The added dependency of removing finalizers only after pod
+recreation [Tracking the number of failures per index](#tracking-the-number-of-failures-per-index)
+may keep pods around longer (around 10s which is the backoff for pod recreation)
+before actual deletion (requested or by PodGC).
```

Comment:
We don't have any SLI/SLO around pod deleteion, so it won't really matter.

It's a useful notice, but I'm not sure it actually fits into this question.

Reply:
Ok, I delete this note from here as this is a copy-paste anyway from this section: https://github.com/kubernetes/enhancements/pull/3967/files#diff-4908561f5b948cdeee23dd68a5a9d77208c89394bb2f05b898a8d679802a8e46R1118-R1123